File: Algorithms/searching/binarysearch2.cpp
#include <bits/stdc++.h>
#include<cstdlib>
#include<time.h>
using namespace std;
int main()
{
  srand(time(0));
  int length=rand()%10;
  int A[length];
  for(int i=0;i<length;i++)
    A[i]=rand()%10;
  int x=rand()%10;
  //I have to sort the array
  sort(A,A+length);
  for(int i=0;i<length;i++)
    cout<<A[i]<<',';
  cout<<"\b \nLet us search for " <<x<<endl;
  int k=0;
  for(int i=length/2;i>=1;i/=2)
    while(k+i<length&&A[k+i]<=x)
      k+=i;
  if(A[k]==x)
    cout<<"Element found";
  else
    cout<<"Element not found";
}

File: Algorithms/searching/binarysearch2.py
import random
A=random.sample(range(10),random.randint(0,10))
x=random.randint(0,10)
A.sort()
print(A)
print("Let us search for "+ str(x))
b=len(A)
k=0
while b>1:
    b=int(b/2)
    while k+b<len(A) and A[k+b]<=x:
        k+=b
if A[k]==x:
    print("Element found")
else:
    print("Element not found")
